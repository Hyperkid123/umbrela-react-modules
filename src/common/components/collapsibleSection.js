import React, {Component} from 'react';
import PropTypes from 'prop-types';
import ExpansionPanel, {
  ExpansionPanelSummary,
  ExpansionPanelDetails,
} from 'material-ui/ExpansionPanel';
import Typography from 'material-ui/Typography';
import ExpandMoreIcon from 'material-ui-icons/ExpandMore';

export default class CollapsibleSection extends Component {
  render() {
    return (
      <ExpansionPanel>
        <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
          <Typography>Expansion Panel 1</Typography>
        </ExpansionPanelSummary>
        <ExpansionPanelDetails>
          <Typography>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex,
            sit amet blandit leo lobortis eget.
          </Typography>
        </ExpansionPanelDetails>
      </ExpansionPanel>
    );
  }
}

CollapsibleSection.propTypes = {
  sectionTitle: PropTypes.string.isRequired,
  children: PropTypes.object,
};
