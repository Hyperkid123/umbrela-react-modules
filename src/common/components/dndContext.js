import {DragDropContext} from 'react-dnd';
import HTML5Backend from 'react-dnd-html5-backend';
import React,{Component} from 'react';

export class DNDContext extends Component {
    render() {
        return this.props.children
    }
}



export default DragDropContext(HTML5Backend)(DNDContext);
