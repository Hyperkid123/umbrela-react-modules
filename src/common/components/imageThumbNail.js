import React from 'react';
import {ThumbNail} from '../styledComponents/containers';

export const ImageThumbNail = ({src, alt}) => (
  <div style={{margin: 5, display: 'inline-block'}}>
    <ThumbNail src={src} alt={alt}/>
  </div>
)
