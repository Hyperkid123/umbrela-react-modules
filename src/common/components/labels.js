import React from 'react';
import {
  LabelContainer,
} from '../styledComponents/containers';

export const TextFieldComent = ({label, alignRight, error}) => (
  <LabelContainer alignRight error={error}>
    {label}
  </LabelContainer>
);
