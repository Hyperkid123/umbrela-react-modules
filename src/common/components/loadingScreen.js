import React from 'react';
import { CircularProgress } from 'material-ui/Progress';
import {Flex} from '../styledComponents/containers';

const LoadingScreen = () => (
    <Flex horizontalCenter grow verticalCenter>
      <CircularProgress size={80} thickness={5} />
    </Flex>
);

export default LoadingScreen;
