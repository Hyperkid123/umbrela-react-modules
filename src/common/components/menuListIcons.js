import React from 'react';
import ExpandLess from 'material-ui-icons/ExpandLess';
import ExpandMore from 'material-ui-icons/ExpandMore';
import IconButton from 'material-ui/IconButton';
import {
  MenuListIconsContainer,
  MenuItem,
  MenuListLabel,
  Flex,
} from '../styledComponents/containers';
import {DefaultLink} from '../styledComponents/typography';
import Tooltip from 'material-ui/Tooltip';

export const ChevronUp = ({tooltip}) => (
  <IconButton tooltip={tooltip}>
    <ExpandLess />
  </IconButton>
);

export const ChevronDown = ({tooltip}) => (
  <IconButton tooltip={tooltip}>
    <ExpandMore />
  </IconButton>
);

export const MenuListIconsWrapper = ({tooltipUp, tooltipDown}) => (
  <MenuListIconsContainer>
    <ChevronUp tooltip={tooltipUp}/>
    <ChevronDown tooltip={tooltipDown}/>
  </MenuListIconsContainer>
)

export const MenuListItem = ({link, label, onClick, sheetId, active, tooltipLabel, dragging, preventDrag}) => (
        <Tooltip
          title={tooltipLabel}
          enterDelay={500}
          disableTriggerFocus={dragging || preventDrag}
          disableTriggerHover={dragging || preventDrag}
          disableTriggerTouch={dragging || preventDrag}
          placement='right-end'
        >
          <MenuItem dragging={dragging} data-tip={tooltipLabel} active={active} onClick={() => onClick(sheetId)}>
          <Flex verticalCenter>
          <MenuListLabel>
            <DefaultLink href={link}>{label}</DefaultLink>
          </MenuListLabel>
        </Flex>
    </MenuItem>
  </Tooltip>
)
