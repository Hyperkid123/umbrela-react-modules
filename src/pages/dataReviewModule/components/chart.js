import React,{Component} from 'react';
//import CloseQuestions from './CloseQuestions';
import OpenQuestion from './openQuestion';
//import MatrixQuestion from './MatrixQuestion';
import PropTypes from 'prop-types';
import {Flex} from '../../../common/styledComponents/containers';
//import LegendTable from './LegendTable';
//import DivideQuestion from './DivideQuestion';

import {
  OPEN_QUESTION,
  CLOSE_QUESTION,
  CLOSE_MULTI_QUESTION,
  CLOSE_WITH_OPEN_QUESTION,
  CLOSE_MULTI_WITH_OPEN_QUESTION,
  MATRIX_MULTI_IMAGE_QUESTION,
  MATRIX_MULTI_QUESTION,
  MATRIX_SINGLE_IMAGE_QUESTION,
  MATRIX_SINGLE_QUESTION,
  DIVIDE_QUESTION,
  ORDER_QUESTION,
  OPEN_WITH_IMAGE_QUESTIONS,
  IMAGE_OPTIONS_QUESTION,
  IMAGE_OPTIONS_MULTI_QUESTION,
  IMAGE_OPTIONS_WITH_OPEN_QUESTION,
  IMAGE_OPTIONS_MULTI_WITH_OPEN_QUESTION,
  MEDIA_QUESTION,
  MEDIA_MULTI_QUESTION,
  MEDIA_WITH_OPEN_QUESTION,
  MEDIA_MULTI_WITH_OPEN_QUESTION,
  isOptionImage,
  hasImagePreview,
} from '../../../common/chartTypes';

export default class Chart extends Component {

  renderChart = () => {
    switch (this.props.questionType) {
      case OPEN_QUESTION:
      case OPEN_WITH_IMAGE_QUESTIONS:
        return (
          <Flex>
            <OpenQuestion question={this.props.question} data={this.props.data}/>
          </Flex>
        );
      case CLOSE_QUESTION:
      case CLOSE_MULTI_QUESTION:
      case CLOSE_WITH_OPEN_QUESTION:
      case CLOSE_MULTI_WITH_OPEN_QUESTION:
      case IMAGE_OPTIONS_QUESTION:
      case IMAGE_OPTIONS_MULTI_QUESTION:
      case IMAGE_OPTIONS_WITH_OPEN_QUESTION:
      case IMAGE_OPTIONS_MULTI_WITH_OPEN_QUESTION:
      case MEDIA_QUESTION:
      case MEDIA_MULTI_QUESTION:
      case MEDIA_WITH_OPEN_QUESTION:
      case MEDIA_MULTI_WITH_OPEN_QUESTION:
        return (
            <Flex>
              {/**<CloseQuestions
                hasImagePreview={hasImagePreview(this.props.question.type)}
                showLegend={this.props.showLegend}
                chartType={this.props.chartType}
                data={this.props.data.barData}
                url={this.props.question.url}
              />
              <LegendTable isOptionImage={isOptionImage(this.props.question.type)} data={this.props.data}/>
            */}
            </Flex>
        );
      case  MATRIX_MULTI_IMAGE_QUESTION:
      case  MATRIX_MULTI_QUESTION:
      case  MATRIX_SINGLE_IMAGE_QUESTION:
      case  MATRIX_SINGLE_QUESTION:
        return (
          <Flex>
            {/**

            <MatrixQuestion
              data={this.props.data}
              chartType={this.props.chartType}
              showLegend={this.props.showLegend}
              question={this.props.question}
              isOptionImage={isOptionImage(this.props.question.type)}
            />
          */}
          </Flex>
        );
      case DIVIDE_QUESTION:
      case ORDER_QUESTION:
        return(
          {/**<DivideQuestion data={this.props.data}/>*/}
        );
      default:
        return null;

    }
  }

  render() {
    return this.renderChart();
  }
}

Chart.propTypes = {
  data: PropTypes.object.isRequired,
  chartType: PropTypes.string.isRequired,
  showLegend: PropTypes.bool.isRequired,
  questionType: PropTypes.string.isRequired,
  question: PropTypes.object.isRequired,
};
