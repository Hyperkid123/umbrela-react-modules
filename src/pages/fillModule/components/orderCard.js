/**
 * Created by rela on 27/06/17.
 */
import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {findDOMNode} from 'react-dom';
import {DragSource, DropTarget} from 'react-dnd';
import flow from 'lodash/flow';
import {
  OrderQuestionItem
} from '../../../common/styledComponents/containers'

import SvgIcon from 'material-ui/SvgIcon';

const cardSource = {
    beginDrag(props){
        return {
            id: props.id,
            index: props.index,
        };
    },
};

const cardTarget = {
    hover(props, monitor, component){
        const dragIndex = monitor.getItem().index;
        const hoverIndex = props.index;
        if (dragIndex === hoverIndex) {
            return;
        }
        const hoverBoundingRect = findDOMNode(component).getBoundingClientRect();
        const hoverMiddleY = (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2;
        const clientOffset = monitor.getClientOffset();
        const hoverClientY = clientOffset.y - hoverBoundingRect.top;
        if (dragIndex < hoverIndex && hoverClientY < hoverMiddleY) {
            return;
        }
        if (dragIndex > hoverIndex && hoverClientY > hoverMiddleY) {
            return;
        }
        props.moveCard(dragIndex, hoverIndex);
        monitor.getItem().index = hoverIndex;
    },
};


const buttonStyle = {
    border: 'none',
    padding: 0,
    marginRight: 5,
};

const previewStyle = {
  display: 'inline-block'
}

class Card extends Component {
    render() {
        const { isDragging, connectDragSource, connectDropTarget, connectDragPreview } = this.props;
        const opacity = isDragging ? 0 : 1;
        return connectDragSource(connectDropTarget(
            <div style={{...opacity, textAlign: 'center'}}>
              {connectDragPreview(
                <span style={previewStyle}>
                  <OrderQuestionItem>
                    <SvgIcon style={buttonStyle} color='disabled'>
                      <path d="M13,11H18L16.5,9.5L17.92,8.08L21.84,12L17.92,15.92L16.5,14.5L18,13H13V18L14.5,16.5L15.92,17.92L12,21.84L8.08,17.92L9.5,16.5L11,18V13H6L7.5,14.5L6.08,15.92L2.16,12L6.08,8.08L7.5,9.5L6,11H11V6L9.5,7.5L8.08,6.08L12,2.16L15.92,6.08L14.5,7.5L13,6V11Z" />
                    </SvgIcon>
                    <span>
                      {this.props.option.optionTitle}
                    </span>
                  </OrderQuestionItem>
                </span>
              )}
            </div>
        ));
    }
}


Card.propTypes = {
    connectDragSource: PropTypes.func.isRequired,
    connectDropTarget: PropTypes.func.isRequired,
    index: PropTypes.number.isRequired,
    isDragging: PropTypes.bool.isRequired,
    id: PropTypes.any.isRequired,
    moveCard: PropTypes.func.isRequired,
    option: PropTypes.object.isRequired,
};

export default flow(DropTarget('CARD', cardTarget, connect => ({
    connectDropTarget: connect.dropTarget(),
})), DragSource('CARD', cardSource, (connect, monitor) => ({
    connectDragSource: connect.dragSource(),
    connectDragPreview: connect.dragPreview(),
    isDragging: monitor.isDragging(),
})))(Card);
