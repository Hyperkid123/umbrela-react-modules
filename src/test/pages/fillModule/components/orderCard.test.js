import React from 'react'
import {Card as Snapshot} from '../../../../pages/fillModule/components/orderCard';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Order card component', () => {
  const initialProps = {};
  beforeEach(() => {
    initialProps.isDragging = false;
    initialProps.connectDragSource = jest.fn();
    initialProps.connectDropTarget = jest.fn();
    initialProps.connectDragPreview = jest.fn();
    initialProps.index = 0;
    initialProps.id = 55;
    initialProps.moveCard = jest.fn();
    initialProps.option = {
      optionTitle: 'foo'
    }
  });

  it('should render correctly', () => {
    const tree = shallow(
        <Snapshot {...initialProps}/>
    )
    expect(toJson(tree)).toMatchSnapshot();
  })
});
