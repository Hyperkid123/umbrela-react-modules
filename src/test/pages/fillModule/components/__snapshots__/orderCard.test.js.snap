// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order card component should render correctly 1`] = `""`;
